app/models/katello/repository.rb

```diff
@@ -515,5 +527,11 @@ def ensure_valid_docker_attributes
         errors.add(:base, N_("Repository URL or Upstream Name is empty. Both are required for syncing from the upstream."))
       end
     end
+
+    def ensure_has_url_for_ostree
+      if url.blank? && library_instance_id.nil?
+        errors.add(field, N_("cannot be blank. RPM OSTree Repository URL required for syncing from the upstream."))
+      end
```

Comment:
After watching a Sandi Metz talk, I have become a fan of using guards in situations like this, e.g.:

```
def ensure_branch_only_for_ostree
  return true if url.blank? && library_instance_id.nil?
  errors.add(field, N_("cannot be blank. RPM OSTree Repository URL required for syncing from the upstream."))
end
```